dvc/repo/experiments/executor/manager.py

```diff
@@ -0,0 +1,340 @@
+
+        from .base import ExecutorInfo
+        from .ssh import SSHExecutor
+
+        def make_executor(info: "ExecutorInfo"):
+            if info.git_url:
+                scheme = urlparse(info.git_url)
+                if scheme == "file":
+                    cls: Type = TempDirExecutor
+                elif scheme == "ssh":
+                    cls = SSHExecutor
+                else:
+                    raise NotImplementedError
+            else:
+                cls = WorkspaceExecutor
+            return cls.from_info(info)
+
+        for name in self.proc:
+            infofile = os.path.join(name, f"{name}{BaseExecutor.INFOFILE_EXT}")
+            try:
+                with open(infofile, encoding="utf-8") as fobj:
+                    info = ExecutorInfo.from_dict(json.load(fobj))
+                yield name, make_executor(info)
+            except OSError:
+                continue
+
+    @classmethod
+    def from_stash_entries(
+        cls,
+        scm: "Git",
+        wdir: str,
+        repo: "Repo",
+        to_run: Dict[str, ExpStashEntry],
+        **kwargs,
+    ):
+        manager = cls(scm, wdir)
+        try:
+            for stash_rev, entry in to_run.items():
+                scm.set_ref(EXEC_HEAD, entry.head_rev)
+                scm.set_ref(EXEC_MERGE, stash_rev)
+                scm.set_ref(EXEC_BASELINE, entry.baseline_rev)
+
+                # Executor will be initialized with an empty git repo that
+                # we populate by pushing:
+                #   EXEC_HEAD - the base commit for this experiment
+                #   EXEC_MERGE - the unmerged changes (from our stash)
+                #       to be reproduced
+                #   EXEC_BASELINE - the baseline commit for this experiment
+                executor = cls.EXECUTOR_CLS.from_stash_entry(
+                    repo,
+                    stash_rev,
+                    entry,
+                    **kwargs,
+                )
+                manager.enqueue(stash_rev, executor)
+        finally:
+            for ref in (EXEC_HEAD, EXEC_MERGE, EXEC_BASELINE):
+                scm.remove_ref(ref)
+        return manager
+
+    def exec_queue(self, jobs: Optional[int] = 1, detach: bool = False):
+        """Run dvc repro for queued executors in parallel."""
+        if detach:
+            raise NotImplementedError
+            # TODO use ProcessManager.spawn() to support detached runs
+        return self._exec_attached(jobs=jobs)
+
+    def _exec_attached(self, jobs: Optional[int] = 1):
+        import signal
+        from concurrent.futures import (
+            CancelledError,
+            ProcessPoolExecutor,
+            wait,
+        )
+        from multiprocessing import Manager
+
+        from dvc.stage.monitor import CheckpointKilledError
+
+        result: Dict[str, Dict[str, str]] = defaultdict(dict)
+
+        manager = Manager()
+        pid_q = manager.Queue()
+
+        with ProcessPoolExecutor(max_workers=jobs) as workers:
```

Comment:
This should eventually go away entirely in favor of having "attached" execution done by `ProcessManager` (or rather via an actual task queue that works on top of `ProcessManager`).

In particular, we should not need to be managing multiprocessing worker signals/PIDs ourselves in exp (and it should all be abstracted away by the process manager).